File: src/schema/application/devices.ts
import { pgTable, text, boolean} from 'drizzle-orm/pg-core';
import * as path from 'path';
import {
  fileRegex,
  getConfigDefaults,
  system_fields,
} from '@dna-platform/crdt-lww-postgres/build/schema/system';
// import { table as addresses } from './addresses';

const filename = path.basename(__filename).replace(fileRegex, '');
const config = getConfigDefaults.byIndex(filename);
export const table = pgTable(
  filename,
  {
    ...system_fields,
    id: text('id').primaryKey(),
    is_traffic_monitoring_enabled: boolean('is_traffic_monitoring_enabled').default(false),
    is_config_monitoring_enabled: boolean('is_config_monitoring_enabled').default(false),
    is_telemetry_monitoring_enabled: boolean('is_telemetry_monitoring_enabled').default(false),
    is_device_authorized: boolean('is_device_authorized').default(false),
    
    device_uuid: text('device_uuid').default(""),
    device_name: text("device_name").default(""),
    device_category: text("device_category").default(""),
    device_type: text("device_type").default(""),
    device_os: text("device_os").default(""),

    is_device_online: boolean("is_device_online").default(false),
  },
  config,
);

File: bin/store/src/database/schema/tables/devices.rs
use crate::define_table_schema;
use crate::generated::schema::generator::diesel_schema_definition::{
    types::*, DieselTableDefinition,
};
use crate::{system_fields, system_foreign_keys, system_indexes};

/**
 * Auto-converted from TypeScript schema source.
 */
pub struct DevicesTable;

define_table_schema! {
    hypertable: false,
    fields: {
        // System fields - common across all tables ( REQUIRED )
        system_fields!(),

        is_traffic_monitoring_enabled: nullable(boolean()), default "false",
        is_config_monitoring_enabled: nullable(boolean()), default "false",
        is_telemetry_monitoring_enabled: nullable(boolean()), default "false",
        is_device_authorized: nullable(boolean()), default "false",
        device_uuid: nullable(text()), default "",
        device_name: nullable(text()), default "",
        device_category: nullable(text()), default "",
        device_type: nullable(text()), default "",
        device_os: nullable(text()), default "",
        device_version: nullable(text()), default "",
        is_device_online: nullable(boolean()), default "false",
        address_id: nullable(text()),
    },
    indexes: {
        // System field indexes ( REQUIRED )
        system_indexes!("devices"),
    },
    foreign_keys: {
        // System field foreign keys ( REQUIRED )
        system_foreign_keys!("devices"),
        fk_devices_address_id: { columns: ["address_id"], foreign_table: "addresses", foreign_columns: ["id"], on_delete: "no action", on_update: "no action" },
    }
}
